crates/trie/parallel/src/proof_task.rs

```diff
@@ -112,14 +111,20 @@ enum StorageWorkerJob {
 /// # Shutdown
 ///
 /// Worker shuts down when the crossbeam channel closes (all senders dropped).
-fn storage_worker_loop<Tx>(
-    proof_tx: ProofTaskTx<Tx>,
+fn storage_worker_loop<Factory>(
+    view: ConsistentDbView<Factory>,
+    task_ctx: ProofTaskCtx,
     work_rx: CrossbeamReceiver<StorageWorkerJob>,
     worker_id: usize,
     #[cfg(feature = "metrics")] metrics: ProofTaskTrieMetrics,
 ) where
-    Tx: DbTx,
+    Factory: DatabaseProviderFactory<Provider: BlockReader>,
 {
+    // Create db transaction before entering work loop
+    let provider = view.provider_ro()
+        .expect("Storage worker failed to initialize: database unavailable");
```

Comment:
Same for 

> instead of having expect 
> 
> https://github.com/paradigmxyz/reth/pull/19012/files#diff-e3fb5658ad9c8b5024a6858cc898487c1961023c6361aeab5b4dcdd7ef781fb6L1231-L1233
> 
> its much simpler to expect here, and we dont have to propogate the error up